### PR TITLE
Rackspace fixes

### PR DESCRIPTION
**Pull request by doublerr**
Please do not merge; looking for comments

This is a pretty large update required to get the cluster running on Rackspace again.

Notable changes:
- Use Flanneld instead of bash for next hop routing
- Use KelseyHightowers kubelet registration tool
- Added in token support for kubelet,kube-proxy <-> apiserver communication
  - Bearer tokens are passed to nodes via cloud-config. I may switch this to use metadata however there isn't an easy way to read this data without another tool.
  - The known_tokens.csv file is passed as file injection via nova. It's then copied from the metadata location to /var/lib/kube-apiserver prior to the unit starting. This isn't ideal so I'm thinking of alternatives.

I hard coded python2.7 in the scripts since the commands don't work in python3. Thoughts?

cc @erictune

### CODE REVIEW

**Comment by doublerr** [comment]
I'm part of the Rackspace Corporation CLA. However the email address on file there differs from my github.

**Comment by erictune** [comment]
FYI once we merge https://github.com/GoogleCloudPlatform/kubernetes/pull/6949 then you won't need KelseyHightowers kubelet registration tool.  Same behavior will be built-in.

File: cluster/rackspace/authorization.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create generic token following GCE standard
+create_token() {
+  echo $(cat /dev/urandom | base64 | tr -d "=+/" | dd bs=32 count=1 2> /dev/null)
+}
+
+get_tokens_from_csv() {
+  KUBE_BEARER_TOKEN=$(awk -F, '/kubelet/ {print $1}' ${KUBE_TEMP}/${1}_tokens.csv)
```
Comment (erictune) [inline cluster/rackspace/authorization.sh]: In GCE and other setups, the kubelet token is called KUBELET_TOKEN and there is a third token, for the human cluster administrator, which is called KUBE_BEARER_TOKEN.  You've mixed these two together. In the future, the kubelet token will imply different privileges than the admin token.  So, future updates will be easier if you follow this pattern now.

Comment (erictune) [inline cluster/rackspace/authorization.sh]: In GCE it says this:
```bash
    (umask 077;
      echo "${KUBE_BEARER_TOKEN},admin,admin" > "${KNOWN_TOKENS_FILE}";
      echo "${KUBELET_TOKEN},kubelet,kubelet" >> "${KNOWN_TOKENS_FILE}";
      echo "${KUBE_PROXY_TOKEN},kube_proxy,kube_proxy" >> "${KNOWN_TOKENS_FILE}")
```

**Comment by erictune** [comment]
Please update docs/getting-started-guides/README,md to show rackspace at v0.16.2.

**Comment by erictune** [comment]
I don't understand the flannel or register parts well enough to give it a really thorough review, but if it is working for you, that's fine with me.  

Rest looks good.

**Comment by WIZARD-CXY** [comment]
@doublerr I am the maintainer of the k8s deployment on ubuntu. I read your code and want to learn the dns and some other service token setup from you to update ubuntu version of deployment. I am using the same approach as you without saltstack.Do you want to include the dns or monitoring?

File: cluster/rackspace/authorization.sh (hunk 1 above)
Comment (doublerr) [inline cluster/rackspace/authorization.sh]: Thanks for the info Eric. I'll update it.

**Comment by doublerr** [comment]
@WIZARD-CXY I do want to include those items but haven't dug into how they're being activated yet. I know we have variables to turn these features off/on but we don't have the deployment mechanisms in place.

**Comment by WIZARD-CXY** [comment]
@doublerr I think I can get the addons done once I undenstand this new feature https://github.com/GoogleCloudPlatform/kubernetes/pull/7101

**Comment by doublerr** [comment]
@erictune this works with 0.16.2 but not with 0.17.0 (I haven't dug into why yet). 

Can we merge to get since readme points to 0.16.2 assuming the bearer items were resolved.

File: cluster/rackspace/authorization.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+get_token_from_csv() {
+  KUBE_BEARER_TOKEN=$(awk -F, '/admin/ {print $1}' ${KUBE_TEMP}/${1}_tokens.csv)
+  KUBELET_TOKEN=$(awk -F, '/kubelet/ {print $1}' ${KUBE_TEMP}/${1}_tokens.csv)
+  KUBE_PROXY_TOKEN=$(awk -F, '/kube_proxy/ {print $1}' ${KUBE_TEMP}/${1}_tokens.csv)
+}
+
+generate_admin_token() {
+  echo "$(create_token),admin,admin" >> ${KUBE_TEMP}/${1}_tokens.csv
```
Comment (erictune) [inline cluster/rackspace/authorization.sh]: I don't get why you have ${1} here.

**Comment by erictune** [comment]
Thanks for the update!

**Comment by erictune** [comment]
sorry for slow review: was out of office last week.

**Pull request merged**